### PR TITLE
fix(Core/CreatureScript): Grand Warlock Nethekurse

### PR DESCRIPTION
**Pull request by Stoabrogga**
##### CHANGES PROPOSED:
- Ensure that Grand Warlock Nethekurse (Shattered Halls) gets attackable even if you pull the Fel Orc Converts far away before killing.
- Fix rescheduling the spell events during battle.
- Fix Dark Spin being correctly cast and damaging random players with the Shadow Bolts.

##### ISSUES ADDRESSED:
none

##### TESTS PERFORMED:
- tested build on Ubuntu 16.04 / clang 7
- tested successfully in-game

##### HOW TO TEST THE CHANGES:
- `.go 183.058 216.841 -20.9767 540`
- fight against Nethekurse

##### KNOWN ISSUES AND TODO LIST:
none

##### Target branch(es):
- [x] Master
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Endeffect-AT** [comment]
Compiled on win 10 pro 64 bit without errors. 

Testet the combat several times after instance reset each time. 

- Grand Warlock Nethekurse can only be attacked AFTER killing the Fel Orc Converts. No matter how far away I pulled them 

- rescheduliung works fine

- The Dark spin casts and damaging random players with the Shadow Bolts as expected

WoWScrnShot_121719_200617

WoWScrnShot_121719_200624

**Comment by Endeffect-AT** [comment]
Sure about this spell is blizzlike?

image

**Comment by Endeffect-AT** [comment]
Edit: Works fine :+1:

**Comment by Helias** [comment]
Good job @Endeffect-AT with all these tests! :+1:

**Comment by Endeffect-AT** [comment]
It's a pleasure! @Helias

File: src/server/scripts/Outland/HellfireCitadel/ShatteredHalls/boss_nethekurse.cpp [hunk 1]
```diff
@@ -264,8 +270,47 @@ class spell_tsh_shadow_sear : public SpellScriptLoader
...
+        class spell_tsh_shadow_bolt_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_tsh_shadow_bolt_SpellScript);
+
+            void SelectRandomPlayer(WorldObject*& target)
+            {
+                if (Creature* caster = GetCaster()->ToCreature())
+                {
+                    std::list<Player*> playerList;
+                    playerList.clear();
```
Comment (GitVerge) [inline src/server/scripts/Outland/HellfireCitadel/ShatteredHalls/boss_nethekurse.cpp]: why clear here?

Comment (Stoabrogga) [reply src/server/scripts/Outland/HellfireCitadel/ShatteredHalls/boss_nethekurse.cpp]: I don't know, took this over from here:
https://github.com/azerothcore/azerothcore-wotlk/blob/e22d78ecd6997bc11f8620b759364f119dac431a/src/server/scripts/Northrend/AzjolNerub/ahnkahet/instance_ahnkahet.cpp#L268-L269
Is it not necessary?

Comment (GitVerge) [reply src/server/scripts/Outland/HellfireCitadel/ShatteredHalls/boss_nethekurse.cpp]: not to me
@GMKyle

Comment (Stoabrogga) [reply src/server/scripts/Outland/HellfireCitadel/ShatteredHalls/boss_nethekurse.cpp]: Just read that it is not necessary, as the default constructor of std::list automatically creates an empty list. I pushed a new commit removing the line.

**Comment by BarbzYHOOL** [comment]
yeah thank you @Endeffect-AT @GitVerge and our wise @Stoabrogga

**Pull request merged**